modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -887,17 +881,13 @@ CompletableFuture<Void> completeReadOnlyTransactionFuture(TxIdAndTimestamp txIdA
         return readOnlyTxFuture;
     }
 
-    private CompletableFuture<Boolean> onLwnChanged(ChangeLowWatermarkEventParameters parameters) {
+    private CompletableFuture<Boolean> onLwmBeforeChange(ChangeLowWatermarkEventParameters parameters) {
```

Comment:
This listener is no longer required. Just use lwm.getLowWatermark()
getLowWatermark() currently uses busy locks, this is pointless and should be removed.

Reply:
Fixed.